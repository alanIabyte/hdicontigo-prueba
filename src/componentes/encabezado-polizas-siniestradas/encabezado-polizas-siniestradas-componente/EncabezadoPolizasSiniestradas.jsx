import React, { useState, useEffect, memo, lazy, Suspense } from "react";
import PropTypes from "prop-types";
import { useCookies } from "react-cookie";
import { useDispatch, useSelector } from "react-redux";
import { useLazyQuery } from "@apollo/react-hooks";
import { loader } from "graphql.macro";
import RegresarIcono from "@material-ui/icons/ChevronLeftRounded";
import {
  EnvolvedorEncabezado,
  TituloContenedor,
  AvatarUsuario,
} from "./EncabezadoPolizasSiniestradas.styled";
import avatar from "../../../recursos/imagenes/default-avatar.png";
import Constantes from "../../../recursos/constantes";
import { BotonIcono } from "../../encabezado/encabezado-componente/Encabezado.styled";

const Menu = lazy(() => import("../../menu"));

const nombreCookie = Constantes.nombreDeCookie;

const OBTENER_PERFIL = loader(
  "../../../graphQL/query/seguridad/obtener_perfil.graphql"
);

const EncabezadoPolizasSiniestradas = ({
  regresar,
  funcionRegresar,
  mostrarMenu,
}) => {
  const [cookie] = useCookies([nombreCookie]);
  const [seAbrioMenu, asignarValorSeAbrioMenu] = useState(false);
  const objetoCookie = cookie[nombreCookie];
  const estadoApp = useSelector((estado) => estado);
  const dispatch = useDispatch();
  const usuario =
    objetoCookie && objetoCookie.NombreAsegurado
      ? objetoCookie.NombreAsegurado
      : "usuario";

  const diccionario = {
    titulo: `Hola, ${usuario}`,
  };
  const usuarioPerfil = objetoCookie && objetoCookie.Usuario;
  const token = objetoCookie && objetoCookie.access_token;
  const [obtenerPerfil, { loading, error, data }] = useLazyQuery(
    OBTENER_PERFIL,
    {
      fetchPolicy: "cache-first",
    }
  );
  const [imagenUsuario, asignarValorImagenUsuario] = useState(avatar);

  useEffect(() => {
    if (!estadoApp.fotoPerfil) {
      obtenerPerfil({
        variables: { token, usuario: usuarioPerfil },
      });
      return;
    }
    asignarValorImagenUsuario(estadoApp.fotoPerfil);
  }, []);

  useEffect(() => {
    if (
      !loading &&
      data &&
      data.obtener_perfil &&
      data.obtener_perfil.completado
    ) {
      const {
        obtener_perfil: {
          dato: { imagenes },
        },
      } = data;
      asignarValorImagenUsuario(
        imagenes && imagenes.length ? imagenes[0].url : avatar
      );
      dispatch({
        type: "AGREGAR",
        valor: imagenes && imagenes.length ? imagenes[0].url : avatar,
        indice: "fotoPerfil",
      });
    }
  }, [data, loading, error]);

  const alAbrirOCerrarMenu = (seAbrio) => {
    asignarValorSeAbrioMenu(seAbrio);
  };

  return (
    <EnvolvedorEncabezado key="EncabezadoPolizas" seAbrioMenu={seAbrioMenu}>
      {regresar ? (
        <BotonIcono type="button" onClick={funcionRegresar}>
          <RegresarIcono id="botonRegresar" fontSize="large" />
        </BotonIcono>
      ) : (
        mostrarMenu && (
          <Suspense fallback={null}>
            <Menu
              alAbrirOCerrarMenu={alAbrirOCerrarMenu}
              abierto={seAbrioMenu}
              absoluto
            />
          </Suspense>
        )
      )}
      <TituloContenedor id="tituloEncabezado">
        <AvatarUsuario src={imagenUsuario} />
        {diccionario.titulo}
      </TituloContenedor>
    </EnvolvedorEncabezado>
  );
};

EncabezadoPolizasSiniestradas.propTypes = {
  regresar: PropTypes.bool,
  funcionRegresar: PropTypes.func,
  mostrarMenu: PropTypes.bool,
};

EncabezadoPolizasSiniestradas.defaultProps = {
  regresar: false,
  funcionRegresar: () => {},
  mostrarMenu: false,
};

export default memo(
  EncabezadoPolizasSiniestradas,
  (prevProps, nextProps) => prevProps.regresar === nextProps.regresar
);
