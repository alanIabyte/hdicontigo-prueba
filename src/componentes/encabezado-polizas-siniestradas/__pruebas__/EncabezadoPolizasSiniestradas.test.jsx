import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { createStore } from "redux";
import reductor from "../../../reductores/index";
import EncabezadoPolizasSiniestradas from "../encabezado-polizas-siniestradas-componente/EncabezadoPolizasSiniestradas";

const deposito = createStore(reductor);

jest.mock("react-cookie", () => ({
  useCookies: jest.fn().mockReturnValue([{}]),
}));

jest.mock("@apollo/react-hooks", () => ({
  useMutation: jest
    .fn()
    .mockReturnValue([{}, { loading: false, called: false }]),
  useQuery: jest.fn().mockReturnValue({}),
  useLazyQuery: jest.fn().mockReturnValue([{}, {}]),
}));

describe("#EncabezadoPolizasSiniestradas", () => {
  it("se despliega correctamente", () => {
    const tree = renderer
      .create(
        <Provider store={deposito}>
          <EncabezadoPolizasSiniestradas />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
