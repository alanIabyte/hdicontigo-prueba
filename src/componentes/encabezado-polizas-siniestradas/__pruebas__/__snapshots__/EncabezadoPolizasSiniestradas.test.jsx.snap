// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#EncabezadoPolizasSiniestradas se despliega correctamente 1`] = `
<div
  className="sc-ikkxIA eXaLCX"
>
  <div
    className="sc-fqkvVR sc-feUZmu wQLUp iAiTCO"
    id="tituloEncabezado"
  >
    <img
      className="sc-fUnMCh kOFFOF"
      src="test-file-stub"
    />
    Hola, usuario
  </div>
</div>
`;
